File: Main/static/JS/game/effects/poison.js
import { DurableEffect } from "./effect.js";


class Poison extends DurableEffect {
    constructor(hero, game_state, caster, power, duration){
        super('Отравление', 'poison.png', hero, game_state, duration, caster)
        this.power = power
    }

    gen_description(){
        return `Отравление [${this.power}] на ${this.duration} ходов (наложено ${this.caster.name})`
    }

    before_move(){
        this.game_state.log(`${this.hero.name} отравлен и потерял ${this.power} hp`)
        this.hero.loose_hp(this.power, this.caster)
        this.decrease_duration()
        this.hero.be_attacked_animation()
    }
}


export { Poison }

File: Main/models/effects/poison.py
from .effect import *


class Poison(DurableEffect):
    name = 'Poison'
    img_src = 'poison.png'
    value = models.IntegerField()

    def gen_description(self, link):
        return f'Отравление [{self.value}] на {self.duration} ходов (наложено {link.caster.name})'

    def before_move(self, link):
        self.decrease_duration(link)
        link.hero.loose_hp(self.value)
        self.get_game_state(link).create_ui_action('damage', link.hero.id)

